### ISSUE TITLE
Does not build in Vs2015&2017

### ISSUE THREAD

**Issue by schmendrick** [dmanning23/MenuBuddySample#1]
Hi, on checkout and trying to build I get several errors => did the nuget packages get updated?

3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TreeTest.cs(38,27,38,31): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.Label'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TreeTest.cs(52,31,52,38): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.Label'
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Input\MouseInputHandler.cs(51,15,51,31): error CS1061: 'InputState' does not contain a definition for 'CheckControllers' and no extension method 'CheckControllers' accepting a first argument of type 'InputState' could be found (are you missing a using directive or an assembly reference?)
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(51,34,51,48): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(58,30,58,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(74,30,74,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(81,30,81,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(97,30,97,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(104,30,104,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(120,30,120,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(127,30,127,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(143,30,143,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(150,30,150,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TouchOptionsScreen.cs(180,11,180,21): error CS1061: 'Image' does not contain a definition for 'Transition' and no extension method 'Transition' accepting a first argument of type 'Image' could be found (are you missing a using directive or an assembly reference?)
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Layouts\Layout.cs(75,41,75,50): error CS1061: 'IClickable' does not contain a definition for 'Clickable' and no extension method 'Clickable' accepting a first argument of type 'IClickable' could be found (are you missing a using directive or an assembly reference?)
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\TextEditTest.cs(22,19,22,27): error CS1729: 'TextEdit' does not contain a constructor that takes 0 arguments
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\SliderTest.cs(30,17,30,22): error CS1729: 'Label' does not contain a constructor that takes 0 arguments
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Widgets\Label\Label.cs(359,17,359,27): error CS1061: 'IFontBuddy' does not contain a definition for 'ScaleToFit' and no extension method 'ScaleToFit' accepting a first argument of type 'IFontBuddy' could be found (are you missing a using directive or an assembly reference?)
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Widgets\Label\Label.cs(364,17,364,28): error CS1061: 'IFontBuddy' does not contain a definition for 'ShrinkToFit' and no extension method 'ShrinkToFit' accepting a first argument of type 'IFontBuddy' could be found (are you missing a using directive or an assembly reference?)
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(39,31,39,42): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(48,27,48,40): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(57,27,57,40): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(66,27,66,41): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(83,38,83,52): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(97,34,97,48): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(111,29,111,43): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ScrollOptionsScreen.cs(125,30,125,44): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\OptionsScreen.cs(41,34,41,46): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\OptionsScreen.cs(46,35,46,48): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\OptionsScreen.cs(49,34,49,45): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\OptionsScreen.cs(52,38,52,44): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\NumEditTest.cs(22,19,22,26): error CS1729: 'NumEdit' does not contain a constructor that takes 0 arguments
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(31,40,31,59): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(38,36,38,50): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(45,33,45,54): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(59,41,59,52): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(73,30,73,43): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(80,31,80,48): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(87,31,87,47): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\MainMenuScreen.cs(94,38,94,44): error CS1503: Argument 1: cannot convert from 'string' to 'MenuBuddy.MenuEntry'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\BigScrollTest.cs(54,33,54,47): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\DropdownTest.cs(22,19,22,35): error CS7036: There is no argument given that corresponds to the required formal parameter 'screen' of 'Dropdown<string>.Dropdown(IScreen)'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\DropdownTest.cs(38,33,38,47): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddySample\MenuBuddySample.SharedProject\ContextMenuTest.cs(22,65,22,79): error CS1503: Argument 2: cannot convert from 'MenuBuddy.FontSize' to 'Microsoft.Xna.Framework.Content.ContentManager'
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Screens\MessageBoxScreen.cs(90,31,90,48): error CS1061: 'FontBuddy' does not contain a definition for 'BreakTextIntoList' and no extension method 'BreakTextIntoList' accepting a first argument of type 'FontBuddy' could be found (are you missing a using directive or an assembly reference?)
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Screens\MessageBoxScreen.cs(91,31,91,41): error CS1579: foreach statement cannot operate on variables of type '?' because '?' does not contain a public definition for 'GetEnumerator'
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Widgets\Background.cs(50,33,50,60): error CS0154: The property or indexer 'IHighlightable.IsHighlighted' cannot be used in this context because it lacks the get accessor
3>C:\work\other\monogame\MenuBuddy\MenuBuddy\MenuBuddy.SharedProject\Widgets\Buttons\BaseButton.cs(122,24,122,40): error CS0428: Cannot convert method group 'HasTimeRemaining' to non-delegate type 'bool'. Did you intend to invoke the method?

**Comment by schmendrick** [comment]
in case anybody wonders how to get the sample running, check out the menubuddy repo, and reference that one in your checked out sample project. then, for the MenuBuddy git repository, check out git# 4879052483079ad04fc9e81dbffccea274760639 (commit from 05.08.2017 10:31:28)); it then will compile and you can run it

**Comment by dmanning23** [comment]
Doh, sorry about that! I accidently checked in a work-in-progress. I was debugging some work being done on the MenuBuddy by removing the nuget package and adding the project directly to the Sample.
I'll fix it right away to use the latest nuget packages.
Cheers!

**Comment by dmanning23** [comment]
Also just a heads up this is a huge refactor if you are using the MenuBuddy on desktop, I switched everything over from DesktopGL to WindowsDX. 
I don't know if everyone knows this, but the ContentManager never cleans up unused resources until the ContentManager object itself is destroyed, which results in content leaks all over the place. I got around this by aggresively recycling ContentManagers in the MenuBuddy, but there is a hard-to-reproduce, game-breaking bug in MonoGame.DesktopGL when you do this technique:
https://github.com/MonoGame/MonoGame/issues/6165
Again, I apologize to anyone using the MenuBuddy on desktop, but DesktopGL is completely broken until that bug is resolved.
